### ISSUE TITLE
copy & checksum in 1 go ?

### ISSUE THREAD

**Issue by Symbiot78** [ncw/rclone#593]
hi

I'm trying to set up rclone to sync my photo library.

what I'm doing atm is using rsnapshot to create daily, weekly, monthly & yearly..

I want to introduce rclone into the mix by having rclone copy the snapshots to acd. 

So far I've got the following commandline:

rclone --config /media/rclone/.rclone.conf --checksum copy /media/source billeder:billeder

But right now I'm unsure if I can do both --checksum & copy... and if I need anything else in the commandline?

basically I want to have rclone check if there's any changes before the copy command.. no need to do any copying if there's no changes.. 

also I'd like it to be all in 1 line since I'm using rclone in a docker container...

**Comment by klauspost** [comment]
--checksum is a parameter,  not an operation, so you can not run it separately anyway. It seems like the command you write does what you want. Is there any problem with it?

Note that --checksum will read the files on both source and destination, which will probably not be much faster than a delete + copy.

**Comment by balazer** [comment]
The rclone copy command by default (without any extra options) will only transfer files that have changed.  It does that check based on file size and modification date, though for Amazon Drive, it's based only on the file size, since modification dates can't be set.  Adding --checksum will additionally check that the MD5 or SHA1 hash matches.  When copying from a local filesystem, --checksum will slow down the operation a lot, since every file in the source path that exists in the destination needs to be fully read to calculate its checksum.  But it gives you more confidence that all of the files have been copied correctly.  For routine copy operations, you probably don't need --checksum.

**Comment by ncw** [comment]
> But right now I'm unsure if I can do both --checksum & copy... and if I need anything else in the commandline?

It looks fine to me.  You could drop the `--checksum` if you want - rclone always checks the checksums of files it transfers which is probably what you are after, rather than checking the checksums of all files.

> basically I want to have rclone check if there's any changes before the copy command.. no need to do any copying if there's no changes..

That is exactly what the `copy` verb does.

**Comment by ncw** [comment]
I'm going to close this now as I think we've answered the question.  Please re-open if you think otherwise!
